### PR TITLE
[resotoclient][feat] Use aiohttp instead of requests

### PR DESCRIPTION
**Pull request by meln1k**
# Description

Requests replaced with aiohttp

# To-Dos

Missing: complete requests removal from the dependencies

- [x] Lint with `nox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: resotoclient/http_client/aiohttp_client.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
+from resotoclient.http_client import AsyncHttpClient
+from resotoclient.http_client import HttpResponse
+from typing import Dict, Optional, Callable, Union, AsyncIterator, Awaitable
+from resotoclient.models import JsValue
+from resotoclient.jwt_utils import encode_jwt_to_headers
+import aiohttp
+import ssl
+from yarl import URL
```
Comment (lloesche) [inline resotoclient/http_client/aiohttp_client.py:8]: Do we need an external dependency for URL parsing? Python comes with urllib.parse which provides split and unsplit functions.

File: resotoclient/ca.py [hunk 2]
```diff
@@ -1,24 +1,19 @@
...
+import aiohttp
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import hashes
 from cryptography.hazmat.primitives import serialization
-from cryptography.x509.oid import NameOID
 from resotoclient.jwt_utils import decode_jwt_from_headers
 from jwt.exceptions import InvalidSignatureError
 import logging
 from logging import Logger
-import certifi
```
Comment (lloesche) [inline resotoclient/ca.py:13]: Without certifi does the client still work for people who don't use the Resoto CA and just install their own AWS or Verisign provided certificate?

File: resotoclient/__init__.py [hunk 3]
```diff
@@ -139,40 +129,29 @@ def _post(
...
+        return self.sync_client.patch(path, json)
 
     def _delete(
         self, path: str, params: Optional[Dict[str, str]] = None
-    ) -> requests.Response:
-        with requests.Session() as s:
-            self._prepare_session(s)
-            return s.delete(self.resotocore_url + path, params=params)
+    ) -> HttpResponse:
+        if not self.sync_client.running():
```
Comment (lloesche) [inline resotoclient/__init__.py:152]: Saw this code so often. Should this be a `@requires_sync_client_running` decorator or maybe `start()` includes the running check so the if could be saved or something along those lines?

File: resotoclient/ca.py [hunk 4]
```diff
@@ -54,46 +56,12 @@ def cert_to_bytes(cert: Certificate) -> bytes:
     return cert.public_bytes(serialization.Encoding.PEM)
 
 
-def write_ca_bundle(
-    cert: Certificate, cert_path: str, include_certifi: bool = True, rename: bool = True
-) -> None:
-    tmp_cert_path = f"{cert_path}.tmp" if rename else cert_path
-    with open(tmp_cert_path, "wb") as f:
-        if include_certifi:
-            f.write(certifi.contents().encode())
```
Comment (lloesche) [inline resotoclient/ca.py:63]: This line is crucial for the client to work with non-Resoto certificates. Maybe I missed it, but I did not see a replacement for it.
In the current code the Resoto CA is just one of hundreds of CAs that the client supports.

**Review by lloesche — CHANGES_REQUESTED** [review]
The thread does not shut down when the signal to shut down comes and makes the code hang indefinitely.

File: resotoclient/http_client/aiohttp_client.py (hunk 1 above)
Comment (meln1k) [reply resotoclient/http_client/aiohttp_client.py:8]: It comes as a part of the AIOhttp library, so we get it for free.

**Review by aquamatthias — APPROVED** [review]
Thanks!

**Review by lloesche — APPROVED** [review]

**Pull request merged**